salt/states/file.py

```diff
@@ -1469,15 +1474,30 @@ def managed(name,
     else:
         ret['pchanges'] = {}
 
+    if show_diff:
+        show_changes = show_diff
+        msg = (
+            'The \'show_diff\' argument to the file.managed state has been '
+            'deprecated, please use \'show_changes\' instead.'
+        )
+        salt.utils.warn_until('Boron', msg)
+    elif not show_diff and show_diff is not None:
+        show_changes = show_diff
+        msg = (
+            'The \'show_diff\' argument to the file.managed state has been '
+            'deprecated, please use \'show_changes\' instead.'
+        )
+        salt.utils.warn_until('Boron', msg)
+
```

Comment:
Can we combine this into a single `if show_diff is not None`? I think having two if statements is redundant.